query-service-impl/src/test/java/org/hypertrace/core/query/service/pinot/MigrationTest.java

```diff
@@ -91,6 +89,63 @@ public void testQuery() {
         executionContext);
   }
 
+  @Test
+  public void testQuerySelectionUsingMapAttributeWithSubPath() {
+    Builder builder = QueryRequest.newBuilder();
+    builder.addSelection(createComplexAttributeExpression("Span.tags", "span.kind"));
+    ViewDefinition viewDefinition = getDefaultViewDefinition();
+    defaultMockingForExecutionContext();
+
+    assertPQLQuery(
+        builder.build(),
+        "Select mapValue(tags__KEYS,'span.kind',tags__VALUES) FROM SpanEventView "
+            + "where "
+            + viewDefinition.getTenantIdColumn()
+            + " = '"
+            + TENANT_ID
+            + "'",
+        viewDefinition,
+        executionContext);
+  }
+
+  @Test
+  public void testQuerySelectionUsingMapAttributeWithSubPathWithColumnAttribute() {
+    Builder builder = QueryRequest.newBuilder();
+    builder.addSelection(createComplexAttributeExpression("Span.id", "span.kind"));
+    ViewDefinition viewDefinition = getDefaultViewDefinition();
+    defaultMockingForExecutionContext();
+
+    assertPQLQuery(
+        builder.build(),
+        "Select span_id FROM SpanEventView "
+            + "where "
+            + viewDefinition.getTenantIdColumn()
+            + " = '"
+            + TENANT_ID
+            + "'",
+        viewDefinition,
+        executionContext);
+  }
+
+  @Test
+  public void testQuerySelectionUsingMapAttributeWithoutSubPath() {
```

Comment:
CC @kotharironak @aaron-steinfeld 
Can you check these three test cases? I guess this was the expected behavior?